sympy/solvers/solveset.py

```diff
@@ -416,7 +419,12 @@ def _solve_trig(f, symbol, domain):
     elif solns is S.EmptySet:
         return S.EmptySet
     else:
-        return ConditionSet(symbol, Eq(f_original, 0), S.Reals)
+        #try solving through solve_decomposition
+        try:
+            result = solve_decomposition(f_original, symbol, domain)
```

Comment:
I don't think it is a good place for solve_decomposition (it is not limited for the trig eq). You must use it in `solveset` method in proper place.